indexer-events/migrations/20230120134210_add_partitions.sql

```diff
@@ -0,0 +1,121 @@
+-- Create a partition to start after next month (2023-02-01 to 2023-03-01)
+CREATE TABLE fungible_token_events_p202302 PARTITION OF fungible_token_events FOR VALUES FROM (16752096000000000000000000000000000) TO (16776288000000000000000000000000000);
+
+COMMIT;
+
+-- Create a check constraint in the old table (2021-01-01 to 2023-02-01)
+ALTER TABLE fungible_token_events_old ADD CONSTRAINT fungible_token_events_old_check_constraint CHECK (event_index >= 16094592000000000000000000000000000 AND event_index < 16752096000000000000000000000000000) NOT VALID;
+
+-- Attach the old table as partition with previous data (2021-01-01 to 2023-02-01)
+ALTER TABLE fungible_token_events ATTACH PARTITION fungible_token_events_old FOR VALUES FROM (16094592000000000000000000000000000) TO (16752096000000000000000000000000000);
+
+-- Now we can drop the old table constraint 
+ALTER TABLE fungible_token_events_old DROP CONSTRAINT fungible_token_events_old_check_constraint;
+-- DONE!!! At this point the table is usable again :)
+
+
+CREATE OR REPLACE FUNCTION fn_partition_by_range(_tbl text, _poldname text, _pnewname text, _pkey text, _start numeric, _end numeric)
+  RETURNS void
+  LANGUAGE plpgsql AS
+$func$
+BEGIN
+	-- Detach old partition
+	EXECUTE 'ALTER TABLE ' || _tbl || ' DETACH PARTITION ' || _poldname;
+	-- Create partition from _start to _end
+	EXECUTE 'CREATE TABLE ' || _pnewname || ' PARTITION OF ' || _tbl || ' FOR VALUES FROM (' || _start || ') TO (' || _end || ')';
+	-- Insert data to partition from _start to _end
+	EXECUTE 'INSERT INTO ' || _pnewname || ' SELECT * FROM ' || _poldname || ' WHERE ' || _pkey || ' >= ' || _start || ' AND ' || _pkey || ' < ' || _end;
+	-- Delete data from old partition from _start to _end
+	EXECUTE 'DELETE FROM ' || _poldname || ' WHERE ' || _pkey || ' >= ' || _start || ' AND ' || _pkey || ' < ' || _end;
+	-- Add check on old partition from 0 to _start
+	EXECUTE 'ALTER TABLE ' || _poldname || ' ADD CONSTRAINT ' || _poldname || '_check_constraint CHECK (' || _pkey || ' >= 0 AND ' || _pkey || ' < ' || _start || ') NOT VALID';
+	-- Re-Attach old partition from 0 to _start
+	EXECUTE 'ALTER TABLE ' || _tbl || ' ATTACH PARTITION ' || _poldname || ' FOR VALUES FROM (0) TO (' || _start || ')';
+	-- Drop the check on old partition
+	EXECUTE 'ALTER TABLE ' || _poldname || ' DROP CONSTRAINT ' || _poldname || '_check_constraint';
+END
+$func$;
+
+-- 2023-01-01 to 2023-02-01
+SELECT * FROM fn_partition_by_range('fungible_token_events', 'fungible_token_events_old', 'fungible_token_events_p202301', 'event_index', 16698528000000000000000000000000000, 16881696000000000000000000000000000);
+-- 2022-12-01 to 2023-01-01
+SELECT * FROM fn_partition_by_range('fungible_token_events', 'fungible_token_events_old', 'fungible_token_events_p202212', 'event_index', 16698528000000000000000000000000000, 16725312000000000000000000000000000);
+
+-- Vacuum the table
+VACUUM fungible_token_events;
```

Comment:
nit: newline at the end of file

```suggestion
VACUUM fungible_token_events;

```

Reply:
Thanks, will do